network/dime_message.ksy

```diff
@@ -0,0 +1,61 @@
+meta:
+  id: dime_message
+  title: DIME (Direct Internet Message Encapsulation) Message
+  license: CC0-1.0
+  endian: be
+doc: |
+  Direct Internet Message Encapsulation (DIME)
+  is an old Microsoft specification for sending and receiving
+  SOAP messages along with additional attachments,
+  like binary files, XML fragments, and even other
+  SOAP messages, using standard transport protocols like HTTP.
+doc-ref: 
+  - http://xml.coverpages.org/draft-nielsen-dime-02.txt
+  - https://docs.microsoft.com/en-us/archive/msdn-magazine/2002/december/sending-files-attachments-and-soap-messages-via-dime
+seq:
+  - id: dime
+    type: message
+types:
+  message:
+    seq:
+    - id: message
+      type: record
+      repeat: eos
+  record:
+    seq:
+    - id: version
```

Comment:
May make sense to extract these fields, forming 2 bytes, into an own subtype.

Reply:
I have thought about it, but I think it makes more sense this way.

The draft doesn't split this header section at all as far as I looked, so i thing it would make it more confusing. (I don't even know how it would be named :smile:, since the lenght fields are part of the header as well it can't be header, so a preheader? part_of_the_header?)

So unless there is some practical reason to extract this part I don't know about, I would prefer not to.